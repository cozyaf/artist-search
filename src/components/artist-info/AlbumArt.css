.AlbumArt {
  border-radius: 25px;
  text-align: center;
  justify-items: center;
  img {
    width: 120px;
    height: 120px;
    background-size: cover;
    margin: auto
  }
  p {
    color: #000;
    overflow-wrap: break-word;
  }
}