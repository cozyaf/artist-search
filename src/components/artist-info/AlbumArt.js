import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import ImageNotFound from '../../images/ImageNotFound.png';
import styles from './AlbumArt.css';

function AlbumArt({ release }) {
  const { title, date, id, artistId } = release;
  const coverArtArchive = release['cover-art-archive'];

  const src = coverArtArchive.front ? `http://coverartarchive.org/release/${id}/front` : ImageNotFound;

  return (
    <Link to={ `/artist/${artistId}/album/${id}`} style={{ textDecoration: 'none' }}>
      <section className={styles.AlbumArt}>
        <img src={src} alt={`${title} Cover`}/>
        <p>{title}</p>
        <p>Release Date: {date}</p>
      </section>
    </Link>
  );
}

AlbumArt.propTypes = {
  release: PropTypes.object.isRequired
};

export default AlbumArt;
