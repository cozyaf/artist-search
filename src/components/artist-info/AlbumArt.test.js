import React from 'react';
import { shallow } from 'enzyme';
import AlbumArt from './AlbumArt';

describe('AlbumArt component', () => {
  it('renders AlbumArt', () => {
    const wrapper = shallow(<AlbumArt release={{ title:'', date:'', id:'', artistId:'', 'cover-art-archive':{} }}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
