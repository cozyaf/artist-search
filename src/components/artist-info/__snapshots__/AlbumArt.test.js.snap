// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumArt component renders AlbumArt 1`] = `
<Link
  style={
    Object {
      "textDecoration": "none",
    }
  }
  to="/artist//album/"
>
  <section
    className="AlbumArt"
  >
    <img
      alt=" Cover"
      src="/path/image"
    />
    <p />
    <p>
      Release Date: 
    </p>
  </section>
</Link>
`;
